sklearn/ensemble/_stacking.py

```diff
@@ -74,6 +74,7 @@ def __init__(
         n_jobs=None,
         verbose=0,
         passthrough=False,
+        raise_error_with_combined_estimators=True,
```

Comment:
I don't think we need a parameter for that.
I would simply remove the check in `_validate_estimators` in `BaseStacking`.

Reply:
> We need to add a test in `test_stacking.py` to make sure that we can use something like `StackingRegressor([LogisticRegression()])` and `StackingClassifier([Ridge()])`.
> 
> We also need to add a changelog entry in the most recent file (v1.2).

Ok, that is done. When I was writing the tests I noticed that `StackingRegressor([LogisticRegression()])` doesn't make sense. A classifier with the targets of a regressor would fail. So I just changed `StackingClassifier` and left `StackingRegressor` as it was originally.

I noticed that there were no tests to check if the error that I'm removing was raising correctly. Should I add a test to check that `StackingRegressor` fails correctly?

Reply:
The tests where StackingRegressor fails correctly is already present in [sklearn/ensemble/tests/test_common.py](https://github.com/scikit-learn/scikit-learn/blob/5f3d1e57a91e7c89fe3485971188f1ecd335f2c1/sklearn/ensemble/tests/test_common.py) `test_ensemble_heterogeneous_estimators_type`.

This test is failing in the continuous integration because we need to remove `StackingClassifier` from it.